scrapy/linkextractors/lxmlhtml.py

```diff
@@ -52,6 +52,12 @@ def _iter_links(self, document):
                     continue
                 yield (el, attrib, attribs[attrib])
 
+    def check_for_invalid_sublink (self, url):
```

Comment:
A better place for it would be a function in scrapy.utils.url (or even in w3lib.url). It should have a docstring which explains the issue, and tests.

LinkExtractor is not the only place which handles relative URLs - it is also done e.g. in response.urljoin and response.follow.